data/src/main/java/com/linkedin/data/codec/symbol/InMemorySymbolTable.java

```diff
@@ -40,7 +40,7 @@ public InMemorySymbolTable(String symbolTableName, List<String> symbols)
 
     for (int i = 0; i < symbols.size(); i++)
     {
-      String symbol = symbols.get(i);
+      String symbol = symbols.get(i).intern();
```

Comment:
I am all for getting rid of memory waste due to data duplication, but I also have to warn that string interning should be applied judiciously, i.e. where it really has a potential to get rid of enough duplicate data to guarantee substantial memory savings and offset the CPU performance loss. So, how big are these symbol tables typically are, and how likely are they to contain duplicates?

Reply:
Memory reduction is not the primary goal here.
The primary goal is reducing hashmap lookup for symbol name -> ordinal to instance lookup (==) instead of equals. This is safe because we are doing it only for strings in the schema effectively.

Also discussed this with @karthikbalasub offline.